src/client/executor.rs

```diff
@@ -100,30 +100,36 @@ impl Client {
                 return Ok(result);
             }
             Err(err) => {
```

Comment:
The changes here are mostly just shuffling the order so that the error mutations are reflected in the error stored in the server description.